src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -0,0 +1,126 @@
+
+        public ObjectManagementService() { }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(RenameRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
+               ConnectionInfo connInfo;
+               try
+               {
+                   if (connectionService.TryFindConnection(
+                           requestParams.OwnerUri,
+                           out connInfo))
+                   {
+                       using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
+                       {
+                           IRenamable renameObject = this.GetSQLRenameObject(requestParams, sqlConn);
+
+                           renameObject.Rename(requestParams.NewName);
+                       }
+                   }
+                   else
+                   {
+                       Logger.Error("The connection could not be found.");
+                       throw new Exception(SR.ErrorConnectionNotFound);
+                   }
+               }
+               catch (Exception e)
+               {
+                   Logger.Error($"An error occurred while renaming {requestParams.UrnOfObject}: " + e);
+                   throw new Exception(SR.ErrorOnRenameOperationMessage, e);
```

Comment:
what's the value of catching your own exception and rethrowing it as an inner?

Reply:
Here I oriented me on [this suggestion](https://github.com/microsoft/sqltoolsservice/pull/1686#discussion_r972225476). For me the point was that I get to know where the error happened. The inner exception is to know the “cause” of the error ([suggestion](https://github.com/microsoft/sqltoolsservice/pull/1686#discussion_r972224684)). Also, I thought of developer perspective it adds additional information to an exception, which is not interesting for a user.

Reply:
Thinking about it we can just remove this to simplify it - if an error occurs then ADS will know it was trying to rename something so can add additional messaging as needed.

Reply:
@Charles-Gagnon do I understand that correctly, that I should remove the Innerexception or do you mean the whole try-catch?

Reply:
The whole try catch, the dispatcher will already log and send the error itself if one occurs.